### PR TITLE
Pull parameter values from SSM

### PR DESCRIPTION
**Pull request by zachmargolis**
We aren't able to read secrets from the ENV so we need to pull them from SSM directly

~This is a WIP I'm looking for specific feedback on:~

- [x] This code relies on side effects by writing to the ENV, should we just bite the bullet now and update the LexisNexis and AAMVA gems to take explicit config objects? <br /> &nbsp; &nbsp; **update** not going to do this now, but we totally should consider it

- [x] This is the first time we have meaningful code shared across our lambda functions, but I'm not sure the right way to do that, since each Lambda is built into its own separate directory and they can't just walk up the directory tree like a normal gem would --- @crissupb can you help me figure out the right way to share our code across the functions? My attempt at adding a `CodeURi` to the `AWSRubySDKLayer` was clearly not it

- [x] If the code looks good, I can go and update the remaining functions

### CODE REVIEW

File: source/proof_address/lib/proof_address.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@
 require 'faraday'
 require 'retries'
 
+require_relative './ssm_helper' if !defined?(IdentityIdpFunctions::SsmHelper)
```
Comment (zachmargolis) [inline source/proof_address/lib/proof_address.rb:6]: this was my guess at "let's hope that the ssm_helper gets copied in to be next to this file"

File: source/proof_address/lib/proof_address.rb [hunk 2]
```diff
@@ -44,17 +48,40 @@ def post_callback(callback_body:)
...
           "Content-Type" => 'application/json',
           "Accept" => 'application/json'
         )
       end
     end
 
+    def api_auth_token
+      @api_auth_token ||= ENV.fetch("IDP_API_AUTH_TOKEN") do
+        ssm_helper.load('address_proof_result_token')
+      end
```
Comment (zachmargolis) [inline source/proof_address/lib/proof_address.rb:61]: the block form here is so we can defer this and *not* call `ssm_helper.load` unless we need to

File: source/template.yaml [hunk 3]
```diff
@@ -588,6 +588,7 @@ Resources:
       CompatibleRuntimes:
         - ruby2.7
       RetentionPolicy: Retain
+      CodeUri: aws-ruby-sdk/lib/
```
Comment (zachmargolis) [inline source/template.yaml:591]: this was definitely not the right way to say "copy the files in this dir to all the others" but I am interested in how to make that happen @crissupb 

The error this results in:

```
    functools.reduce(lambda message, error: message + " " + str(error), e.causes, str(e))
samcli.commands.validate.lib.exceptions.InvalidSamDocumentException: [InvalidResourceException('AWSRubySDKLayer', 'property CodeUri not defined for resource of type AWS::Serverless::LayerVersion')] ('AWSRubySDKLayer', 'property CodeUri not defined for resource of type AWS::Serverless::LayerVersion')
```

**Review by mitchellhenke — COMMENTED** [review]
Looks good to me, but will defer to @crissupb on approval 🙂

**Review by mitchellhenke — APPROVED** [review]
👍🏼 👍🏼 👍🏼

**Pull request merged**